src/main/java/com/ciandt/techgallery/service/impl/UserServiceTGImpl.java

```diff
@@ -444,4 +477,42 @@ public TechGalleryUser getUserByGoogleId(String googleId) {
     return userDao.findByGoogleId(googleId);
   }
 
+  @Override
+  public TechGalleryUser saveUserPreference(Boolean postGooglePlusPreference, User user)
+      throws NotFoundException, BadRequestException, InternalServerErrorException, IOException, OAuthRequestException {
+
+    validateUser(user);
+
+    TechGalleryUser techUser = userDao.findByEmail(user.getEmail());
+    if (postGooglePlusPreference != null) {
+      techUser.setPostGooglePlusPreference(postGooglePlusPreference);
+      userDao.update(techUser);
+    }
+    return techUser;
+  }
+
+  /**
+   * Validate the user logged in.
+   *
+   * @param user
+   *          info about user from google
+   * @throws InternalServerErrorException
+   *           in case something goes wrong
+   * @throws NotFoundException
+   *           in case the information are not founded
+   * @throws BadRequestException
+   *           in case a request with problem were made.
+   */
+  private void validateUser(User user) throws BadRequestException, NotFoundException, InternalServerErrorException {
```

Comment:
@tsr-thulio GooglePlusCommunicationServiceImpl has the same method. Should it reuse from this?